font/src/darwin/mod.rs

```diff
@@ -454,6 +454,19 @@ impl Font {
     }
 
     pub fn get_glyph(&self, character: char, _size: f64, use_thin_strokes: bool) -> Result<RasterizedGlyph, Error> {
+        // Render custom symbols for underline and beam cursor
+        if character == super::UNDERLINE_CURSOR_CHAR {
```

Comment:
This will be slightly more efficient to use a match. For instance,

```rust
match character {
    UNDERLINE_CURSOR_CHAR => {
    },
    BEAM_CURSOR_CHAR => {
    },
    _ => {
        // default
    }
}
```